src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -444,5 +450,778 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
         {
             return Convert.DefaultToType((IConvertible)this, type, provider);
         }
+
+#if FEATURE_GENERIC_MATH
+        //
+        // IAdditionOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static double IAdditionOperators<double, double, double>.operator +(double left, double right)
+            => left + right;
+
+        // [RequiresPreviewFeatures]
+        // static checked double IAdditionOperators<double, double, double>.operator +(double left, double right)
+        //     => checked(left + right);
+
+        //
+        // IAdditiveIdentity
+        //
+
+        [RequiresPreviewFeatures]
+        static double IAdditiveIdentity<double, double>.AdditiveIdentity => 0.0;
+
+        //
+        // IBinaryNumber
+        //
+
+        [RequiresPreviewFeatures]
+        static bool IBinaryNumber<double>.IsPow2(double value)
+        {
+            var bits = (ulong)BitConverter.DoubleToInt64Bits(value);
+
+            var exponent = (uint)(bits >> ExponentShift) & ShiftedExponentMask;
+            var significand = bits & SignificandMask;
+
+            return (value > 0)
+                && (exponent != MinExponent) && (exponent != MaxExponent)
+                && (significand == MinSignificand);
+        }
+
+        [RequiresPreviewFeatures]
+        static double IBinaryNumber<double>.Log2(double value)
+            => Math.Log2(value);
+
+        //
+        // IBitwiseOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static double IBitwiseOperators<double, double, double>.operator &(double left, double right)
+        {
+            var bits = BitConverter.DoubleToInt64Bits(left) & BitConverter.DoubleToInt64Bits(right);
```

Comment:
Nit: can we not use var here?  Yes, "Int64" is part of the method names on the right, but it still requires some amount of reading and analysis of the right-hand side to figure out what the type here actually is.  I'd find it a whole lot easier to read if it were `long` instead of `var`.

Reply:
Fixed.